plugins/utils/dashboard.nix

```diff
@@ -6,141 +6,378 @@
   ...
 }:
 with lib;
-let
-  cfg = config.plugins.dashboard;
-in
-{
-  options = {
-    plugins.dashboard = {
-      enable = mkEnableOption "dashboard";
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "dashboard";
+  originalName = "dashboard-nvim";
+  defaultPackage = pkgs.vimPlugins.dashboard-nvim;
 
-      package = helpers.mkPluginPackageOption "dashboard" pkgs.vimPlugins.dashboard-nvim;
+  maintainers = [ helpers.maintainers.MattSturgeon ];
 
-      header = mkOption {
-        description = "Header text";
-        type = types.nullOr (types.listOf types.str);
-        default = null;
-      };
+  # TODO introduced 2024-05-30: remove 2024-09-01
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "dashboard"
+      ];
+    in
+    [
+      (mkRemovedOptionModule (
+        basePluginPath ++ [ "sessionDirectory" ]
+      ) "This plugin no longer has session support.")
+    ]
+    ++ (mapAttrsToList
+      (
+        old: new:
+        mkRenamedOptionModule (basePluginPath ++ [ old ]) (basePluginPath ++ [ "settings" ] ++ new)
+      )
+      {
+        header = [
+          "config"
+          "header"
+        ];
+        footer = [
+          "config"
+          "footer"
+        ];
+        center = [
+          "config"
+          "center"
```

Comment:
This should be `center` -> `settings.config.shortcut`, because the the new "center" option only relates to the non-default "doom" theme, while the "shortcut" option is the equivalent for the default "hyper" theme.